components/layout/display_list_builder.rs

```diff
@@ -673,6 +697,35 @@ fn convert_gradient_stops(gradient_items: &[GradientItem],
     stops
 }
 
+#[inline]
+/// Duplicate the first and last stops if necessary.
+///
+/// Explanation by pyfisch:
+/// If the last stop is at the same position as the previous stop the
+/// last color is ignored by webrender. This differs from the spec
+/// (I think so). The  implementations of Chrome and Firefox seem
+/// to have the same problem but work fine if the position of the last
+/// stop is smaller than 100%. (Otherwise they ignore the last stop.)
+///
+/// Similarly the first stop is duplicated if it is not placed
+/// at the start of the virtual gradient ray.
```

Comment:
Thanks for the comment, but... I guess the question is, what does the spec say? :)